### ISSUE TITLE
Popup window width not correctly set when 'signcolumn' and/or 'number' enabled

### ISSUE THREAD

**Issue by bfrg** [vim/vim#6543]
**Describe the bug**
When no `maxwidth` entry is specified for a popup window, Vim won't set the correct width for the popup when the `signcolumn` and/or `number` columns are enabled. The width should be equal to the longest visible line.

**To Reproduce**
1. Run `vim --clean -S test.vim` with
   ```vim
   let winid = popup_create('The quick brown fox jumps over the lazy dog.', {
           \ 'wrap': 0,
           \ 'filter': {-> 0}
           \ })

   call setwinvar(winid, '&number', 1)
   call setwinvar(winid, '&signcolumn', 'yes')
   ```
2. Only `The quick brown fox jumps over the laz` is visible, instead of the full text. 6 characters are missing, 2 for the sign column and 4 for the number column.

**Expected behavior**
When no `maxwidth` is specified, the width of the popup window should be equal to the longest visible line in the popup window, independent of whether sign/number columns are enabled.

**Environment**
Vim version `8.2.1287`

**Comment by bfrg** [comment]
There's another small issue related to the width. Either the documentation or the implementation is wrong.

The documentation says that the width of the popup window is set to the longest line in the buffer (see `:h popup-position`):
https://github.com/vim/vim/blob/6d585f4c5c8d120f768dd61bfd32f6e57ad562a7/runtime/doc/popup.txt#L77-L79

But my observation was that the width is always set to the longest _currently visible_ line in the popup window when no `maxwidth` entry was specified. Example:
```vim
call range(1,10)
        \ ->map({_, i -> string(i)})
        \ ->add('The quick brown fox jumps over the lazy dog.')
        \ ->popup_menu({'wrap': 0, 'maxheight': 3})
```

**Comment by brammool** [comment]
It's indeed the longest visible line that matters.  In most cases all lines will be visible, then it doesn't matter.  In case some lines are hidden the code just looks at the visible lines instead of going through all of them, (which could be slow in case of a long buffer).
A disadvantage is that the width changes when scrolling, but I also think that's doesn't happen very often.

**Comment by bfrg** [comment]
Thank you. Fixed in 0aac67a4314d72a29d3fbee91b6f0ba89e950462.
